### PR TITLE
Merging comparisons with 'in'

### PR DESCRIPTION
**Pull request by edgarriba**
### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>

  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?

</details>

### CODE REVIEW

**Review by cclauss — APPROVED** [review]
LGTM

The quotes in the AssertionErrors a 'bit' 'special' but the strings will concatenate properly.

File: kornia/augmentation/base.py [hunk 1]
```diff
@@ -274,7 +274,7 @@ def __check_batching__(self, input: TensorWithTransformMat):
                     raise AssertionError(
                         f'In batch dimension, input has {inp.shape[0]}' f'but transformation matrix has {mat.shape[0]}'
```
Comment (edgarriba) [inline kornia/augmentation/base.py:275]:
```suggestion
                        f"In batch dimension, input has {inp.shape[0]} but transformation matrix has {mat.shape[0]}"
```

File: kornia/augmentation/base.py [hunk 2]
```diff
@@ -427,7 +427,7 @@ def __check_batching__(self, input: TensorWithTransformMat):
...
                     )
-            elif len(inp.shape) == 3 or len(inp.shape) == 4:
+            elif len(inp.shape) in (3, 4):
                 if len(mat.shape) != 2:
                     raise AssertionError('Input tensor is in non-batch mode ' 'but transformation matrix is not')
```
Comment (edgarriba) [inline kornia/augmentation/base.py:432]:
```suggestion
                    raise AssertionError("Input tensor is in non-batch mode but transformation matrix is not")
```

File: kornia/augmentation/base.py [hunk 3]
```diff
@@ -471,7 +471,7 @@ def __check_batching__(self, input: TensorWithTransformMat):
                     raise AssertionError(
                         f'In batch dimension, input has {inp.shape[0]}' f'but transformation matrix has {mat.shape[0]}'
                     )
-            elif len(inp.shape) == 3 or len(inp.shape) == 2:
+            elif len(inp.shape) in (2, 3):
                 if len(mat.shape) != 2:
                     raise AssertionError('Input tensor is in non-batch mode ' 'but transformation matrix is not')
```
Comment (edgarriba) [inline kornia/augmentation/base.py:476]:
```suggestion
                    raise AssertionError("Input tensor is in non-batch mode but transformation matrix is not")
```
